waltid-services/waltid-wallet-api/src/main/kotlin/id/walt/webwallet/web/controllers/exchange/ExchangeExternalSignaturesController.kt

```diff
@@ -355,6 +350,176 @@ fun Application.exchangeExternalSignatures() = walletRoute {
+                                    IssuanceService.OfferedCredentialProofOfPossessionParameters(
+                                        OfferedCredential(
+                                            format = CredentialFormat.mso_mdoc,
+                                        ),
+                                        ProofOfPossessionParameters(
+                                            ProofType.cwt,
+                                            "<<JSON-ENCODED BYTE ARRAY OF CBOR MAP>>".toJsonElement(),
+                                            "<<JSON-ENCODED BYTE ARRAY OF CBOR MAP>>".toJsonElement(),
+                                        )
+                                    )
+                                ),
+                                credentialIssuer = "https://issuer.portal.walt.id"
+                            )
+                        }
+                        example("When proofType == jwt") {
+                            value = PrepareOID4VCIResponse(
+                                did = "did:web:walt.id",
+                                offerURL = "openid-credential-offer://?credential_offer=",
+                                offeredCredentialsProofRequests = listOf(
+                                    IssuanceService.OfferedCredentialProofOfPossessionParameters(
+                                        OfferedCredential(
+                                            format = CredentialFormat.jwt_vc_json,
+                                        ),
+                                        ProofOfPossessionParameters(
+                                            ProofType.jwt,
+                                            "<<JWT HEADER SECTION>>".toJsonElement(),
+                                            "<<JWT CLAIMS SECTION>>".toJsonElement(),
+                                        )
+                                    )
+                                ),
+                                credentialIssuer = "https://issuer.portal.walt.id"
+                            )
+                        }
+                    }
+                }
+            }
+        }) {
+            val walletService = getWalletService()
+            val req = call.receive<PrepareOID4VCIRequest>()
+            val offer = req.offerURL
+            logger.debug { "Request: $req" }
+
+            val did = req.did ?: walletService.listDids().firstOrNull()?.did
+            ?: throw IllegalArgumentException("No DID to use supplied and no DID was found in wallet.")
+
+            //this can't fail due to the above block
+            val walletDID = DidsService.get(walletService.walletId, did)!!
+            logger.debug { "Retrieved wallet DID: $walletDID" }
+            val authKeyId = ExchangeUtils.getFirstAuthKeyIdFromDidDocument(walletDID.document)
+            logger.debug { "Resolved authorization keyId: $authKeyId" }
+
+            runCatching {
```

Comment:
this could probably be simplified to smth. like:
```kotlin
runCatching {
    WalletServiceManager.externalSignatureClaimStrategy.prepareCredentialClaim(
        did = walletDID.did,
        keyId = authKeyId,
        offerURL = offer,
    )
}.getOrElse { throw IllegalStateException("") }.let { prepareClaimResult ->
    val responsePayload = PrepareOID4VCIResponse(
        did = walletDID.did,
        offerURL = req.offerURL,
        accessToken = prepareClaimResult.accessToken,
        offeredCredentialsProofRequests = prepareClaimResult.offeredCredentialsProofRequests,
        credentialIssuer = prepareClaimResult.resolvedCredentialOffer.credentialIssuer,
    )
    context.respond(HttpStatusCode.OK, responsePayload)
}
```